pyvrp/tests/helpers.py

```diff
@@ -1,30 +1,50 @@
 import pathlib
 import time
 from functools import lru_cache
-from typing import List
+from typing import List, Optional
 
-from pyvrp import ProblemData, Solution, VehicleType
+from pyvrp import Client, ProblemData, Solution, VehicleType
 from pyvrp.read import read as _read
 from pyvrp.read import read_solution as _read_solution
 
 
-def make_heterogeneous(data: ProblemData, vehicle_types: List[VehicleType]):
+def customise(
```

Comment:
Removing `make_heterogeneous` was too much work. I instead replaced it with a `customise` function to make it more general.

Reply:
I'll do this in another PR if you think it's too much work.

Reply:
I think many tests can be rewritten without creating new data files. I'll try rewriting the tests.

Just to understand: why do you prefer to create new data files over a customize function?

Reply:
We've built a lot around the idea that the instance is immutable: once created, the `ProblemData` instance is _it_ - it never changes again. Using the `customise` function, that's still technically true, but only because we're actively working around this restriction by recreating the instance with different properties.

Using the `customise` functions means the test basically applies a diff to the original read instance. We have to keep track of "this part is the base data read from file" and "this is the stuff we apply on top of that". Not really an issue when we do it just once like for the vehicle types, but it's a slippery slope towards more and more parts that can be customised. Then, we're not far off from "why is my test failing and what's the actual instance we're solving here". I'd rather avoid that by not doing this at all. Then the only source of truth is the actual data file, which is easy to fully understand: just read the file.